File: 102_Triangle_Containment/Solution.py
def area(x1, y1, x2, y2, x3, y3):
    return abs((x1*(y2-y3) + x2*(y3-y1) + x3*(y1-y2)) / 2.0)

n = int(input().strip())
count = 0
for _ in range(n):
    
    x1, y1, x2, y2, x3, y3 = map(int, input().split())
    triangle1_area = area(0, 0, x2, y2, x3, y3)
    triangle2_area = area(x1, y1, 0, 0, x3, y3)
    triangle3_area = area(x1, y1, x2, y2, 0, 0)
    
    triangle_area = area(x1, y1, x2, y2, x3, y3)
    
    if triangle_area == triangle1_area + triangle2_area + triangle3_area:
        count += 1

print(count)

File: 102_Triangle_Containment/Solution.go
package main

import (
	"fmt"
	"math"
)

func area(x1, y1, x2, y2, x3, y3 int) float64 {
	return math.Abs(float64((x1*(y2-y3) + x2*(y3-y1) + x3*(y1-y2))) / 2.0)
}

func main() {
	var n, x1, y1, x2, y2, x3, y3 int
	fmt.Scan(&n)

	count := 0
	for i := 0; i < n; i++ {
		fmt.Scan(&x1, &y1, &x2, &y2, &x3, &y3)

		triangle1Area := area(0, 0, x2, y2, x3, y3)
		triangle2Area := area(x1, y1, 0, 0, x3, y3)
		triangle3Area := area(x1, y1, x2, y2, 0, 0)

		triangleArea := area(x1, y1, x2, y2, x3, y3)

		if triangleArea == triangle1Area+triangle2Area+triangle3Area {
			count++
		}
	}

	fmt.Println(count)
}
